### DISCUSSION TITLE
TPM2 Support

### DISCUSSION THREAD

**Discussion by stv0g** [getsops/sops#1321] [Ideas]
I am surprised to see that nobody seems to use SOPS with hardware-backed keys in a TPM2 module.

Is this supported? Or what would be the required steps to do so?

Here are some of my initial ideas:

- Support it via [age-plugin-tpm](https://github.com/Foxboron/age-plugin-tpm)
  - Probably not possible for the same reason as the support of [age-plugin-yubikey](https://github.com/str4d/age-plugin-yubikey) (se https://github.com/getsops/sops/issues/1103)
  
- Support it via GnuPG (>= v2.3)
  - GnuPG seems to have decent support TPMs [as covered in this blog article](https://gnupg.org/blog/20210315-using-tpm-with-gnupg-2.3.html)
  - Use the fallback to gnupg CLI instead of use the Go implementation

**Comment by Foxboron** [comment]
Can just implement something using the TPM2 TSS key files directly.

https://github.com/Foxboron/go-tpm-keyfiles

https://www.hansenpartnership.com/draft-bottomley-tpm2-keys.html


Judging by the existing keysource implementations it should be fairly trivial to implement?

**Comment by Foxboron** [comment]
With https://github.com/getsops/sops/pull/1641 merged you can now probably use https://github.com/Foxboron/age-plugin-tpm in the future.
